### PR TITLE
Update windows ruby version to 3.0

### PR DESCRIPTION
**Pull request by adfoster-r7**
Continuation of https://github.com/rapid7/metasploit-vagrant-builders/pull/18

Updating to compile Ruby 3.0 on the Windows builder machine

### CODE REVIEW

File: scripts/windows/installs/ruby_devkit.ps1 [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# Exit if a cmdlet fails
```
Comment (adfoster-r7) [inline scripts/windows/installs/ruby_devkit.ps1:1]: Swapped this to use powershell

```
Wed Nov 22 16:49:54 GMT 2023 [1;32m==> win-builder-base.amazon-ebs.metasploit-windows-builder: Provisioning with powershell script: scripts/windows/installs/ruby_devkit.bat[0m
Wed Nov 22 16:50:08 GMT 2023 [0;32m    win-builder-base.amazon-ebs.metasploit-windows-builder: a .[0m
Wed Nov 22 16:50:08 GMT 2023 [0;32m    win-builder-base.amazon-ebs.metasploit-windows-builder: a ./devkit.rb[0m
Wed Nov 22 16:50:08 GMT 2023 [0;32m    win-builder-base.amazon-ebs.metasploit-windows-builder: a ./irbrc_predefiner.rb[0m
Wed Nov 22 16:50:08 GMT 2023 [0;32m    win-builder-base.amazon-ebs.metasploit-windows-builder: a ./LICENSE.txt[0m
Wed Nov 22 16:50:08 GMT 2023 [0;32m    win-builder-base.amazon-ebs.metasploit-windows-builder: a ./ruby_installer[0m
Wed Nov 22 16:50:08 GMT 2023 [0;32m    win-builder-base.amazon-ebs.metasploit-windows-builder: a ./ruby_installer.rb[0m
Wed Nov 22 16:50:08 GMT 2023 [0;32m    win-builder-base.amazon-ebs.metasploit-windows-builder: a ./ruby_installer/runtime[0m
Wed Nov 22 16:50:08 GMT 2023 [0;32m    win-builder-base.amazon-ebs.metasploit-windows-builder: a ./ruby_installer/runtime.rb[0m
Wed Nov 22 16:50:08 GMT 2023 [0;32m    win-builder-base.amazon-ebs.metasploit-windows-builder: a ./ruby_installer/runtime/colors.rb[0m
Wed Nov 22 16:50:08 GMT 2023 [0;32m    win-builder-base.amazon-ebs.metasploit-windows-builder: a ./ruby_installer/runtime/components[0m
Wed Nov 22 16:50:08 GMT 2023 [0;32m    win-builder-base.amazon-ebs.metasploit-windows-builder: a ./ruby_installer/runtime/components_installer.rb[0m
Wed Nov 22 16:50:08 GMT 2023 [0;32m    win-builder-base.amazon-ebs.metasploit-windows-builder: a ./ruby_installer/runtime/dll_directory.rb[0m
Wed Nov 22 16:50:08 GMT 2023 [0;32m    win-builder-base.amazon-ebs.metasploit-windows-builder: a ./ruby_installer/runtime/msys2_installation.rb[0m
Wed Nov 22 16:50:08 GMT 2023 [0;32m    win-builder-base.amazon-ebs.metasploit-windows-builder: a ./ruby_installer/runtime/os_process.rb[0m
Wed Nov 22 16:50:08 GMT 2023 [0;32m    win-builder-base.amazon-ebs.metasploit-windows-builder: a ./ruby_installer/runtime/package_version.rb[0m
Wed Nov 22 16:50:08 GMT 2023 [0;32m    win-builder-base.amazon-ebs.metasploit-windows-builder: a ./ruby_installer/runtime/ridk.rb[0m
Wed Nov 22 16:50:08 GMT 2023 [0;32m    win-builder-base.amazon-ebs.metasploit-windows-builder: a ./ruby_installer/runtime/singleton.rb[0m
Wed Nov 22 16:50:08 GMT 2023 [0;32m    win-builder-base.amazon-ebs.metasploit-windows-builder: a ./ruby_installer/runtime/components/01_msys2.rb[0m
Wed Nov 22 16:50:08 GMT 2023 [0;32m    win-builder-base.amazon-ebs.metasploit-windows-builder: a ./ruby_installer/runtime/components/02_pacman_update.rb[0m
Wed Nov 22 16:50:08 GMT 2023 [0;32m    win-builder-base.amazon-ebs.metasploit-windows-builder: a ./ruby_installer/runtime/components/03_dev_tools.rb[0m
Wed Nov 22 16:50:08 GMT 2023 [0;32m    win-builder-base.amazon-ebs.metasploit-windows-builder: a ./ruby_installer/runtime/components/base.rb[0m
Wed Nov 22 16:50:08 GMT 2023 [0;32m    win-builder-base.amazon-ebs.metasploit-windows-builder: tar.exe: could not chdir to 'C:toolsruby26librubysite_ruby2.6.0'[0m
Wed Nov 22 16:50:08 GMT 2023 [0;32m    win-builder-base.amazon-ebs.metasploit-windows-builder:[0m
Wed Nov 22 16:50:08 GMT 2023 [0;32m    win-builder-base.amazon-ebs.metasploit-windows-builder: C:toolsruby26binridk.cmd : The term 'C:toolsruby26binridk.cmd' is not recognized as the name of a cmdlet,[0m
Wed Nov 22 16:50:08 GMT 2023 [0;32m    win-builder-base.amazon-ebs.metasploit-windows-builder: function, script file, or operable program. Check the spelling of the name, or if a path was included, verify that the[0m
Wed Nov 22 16:50:08 GMT 2023 [0;32m    win-builder-base.amazon-ebs.metasploit-windows-builder: path is correct and try again.[0m
Wed Nov 22 16:50:08 GMT 2023 [0;32m    win-builder-base.amazon-ebs.metasploit-windows-builder: At C:WindowsTempscript-655e30af-aab2-d0f4-180a-3336593c20cd.ps1:5 char:1[0m
Wed Nov 22 16:50:08 GMT 2023 [0;32m    win-builder-base.amazon-ebs.metasploit-windows-builder: + C:toolsruby26binridk.cmd install 3[0m
Wed Nov 22 16:50:08 GMT 2023 [0;32m    win-builder-base.amazon-ebs.metasploit-windows-builder: + ~~~~~~~~~~~~~~~~~~~~~~~~~~~~[0m
Wed Nov 22 16:50:08 GMT 2023 [0;32m    win-builder-base.amazon-ebs.metasploit-windows-builder:     + CategoryInfo          : ObjectNotFound: (C:toolsruby26binridk.cmd:String) [], CommandNotFoundException[0m
Wed Nov 22 16:50:08 GMT 2023 [0;32m    win-builder-base.amazon-ebs.metasploit-windows-builder:     + FullyQualifiedErrorId : CommandNotFoundException[0m
Wed Nov 22 16:50:08 GMT 2023 [0;32m    win-builder-base.amazon-ebs.metasploit-windows-builder:[0m
Wed Nov 22 16:50:08 GMT 2023 [0;32m    win-builder-base.amazon-ebs.metasploit-windows-builder: Refreshing environment variables from registry for cmd.exe. Please wait...[0m
Wed
```

File: resources/windows/windows.pkr.hcl [hunk 2]
```diff
@@ -66,8 +101,9 @@ build {
     for_each = local.win_factory.images
     labels   = ["amazon-ebs.win-source"]
     content {
-      ami_name = source.value.name
-      name     = source.value.name
+      # Specify the the AMI name, i.e. metasploit-windows-builder-1.0.8
+      ami_name = "${source.value.name}-${source.value.version}"
```
Comment (adfoster-r7) [inline resources/windows/windows.pkr.hcl:105]: This allows us to build new versions of the AMI without replacing the existing working versions

File: resources/windows/windows.pkr.hcl
Comment (adfoster-r7) [inline resources/windows/windows.pkr.hcl:1]: TODO: We should look at implementing a better pattern than using:

```
./resources/windows/windows.pkr.hcl
48:  force_deregister      = true
49:  force_delete_snapshot = true
```

Probably make it user configurable

```
./resources/windows/windows.pkr.hcl
48:  force_deregister      = "${vars.force_deregister}"
49:  force_delete_snapshot = "${vars.force_delete_snapshot}"
```

And then let the user set the value on the command line:

```
packer build -var "force_deregister=true" -var "force_delete_snapshot=true" resources/windows/windows.pkr.hcl
```

File: scripts/windows/configs/prep_omnibus.bat [hunk 3]
```diff
@@ -1,6 +1,7 @@
 REM Prepare ruby default gemset to handle metasploit omnibus build
-git clone https://github.com/rapid7/metasploit-omnibus
+git clone https://github.com/adfoster-r7/metasploit-omnibus
```
Comment (adfoster-r7) [inline scripts/windows/configs/prep_omnibus.bat:2]: We'll want to get this swapped back

**Review by dwelch-r7 — APPROVED** [review]
LGTM with the caveat that we need from your repo back to that rapid7 owned one

**Pull request merged**